plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteConnectorFactory.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import com.google.inject.Injector;
+import io.airlift.bootstrap.Bootstrap;
+import io.trino.plugin.base.CatalogName;
+import io.trino.plugin.jdbc.ExtraCredentialsBasedIdentityCacheMappingModule;
+import io.trino.plugin.jdbc.JdbcConnector;
+import io.trino.plugin.jdbc.credential.CredentialProviderModule;
+import io.trino.spi.NodeManager;
+import io.trino.spi.classloader.ThreadContextClassLoader;
+import io.trino.spi.connector.Connector;
+import io.trino.spi.connector.ConnectorContext;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spi.type.TypeManager;
+
+import java.util.Map;
+
+import static io.trino.plugin.base.Versions.checkSpiVersion;
+import static java.util.Objects.requireNonNull;
+
+public class IgniteConnectorFactory
```

Comment:
Can we reuse `JdbcConnectorFactory`?

Reply:
``JdbcConnectorFactory`` installed the ``JdbcModule`` by default, Ignite require customed metadata. We may need to overwrite the ``create`` method in ``JdbcConnectorFactory``?

Reply:
Please notice in `JdbcModule`: 
```
        newOptionalBinder(binder, JdbcMetadataFactory.class).setDefault().to(DefaultJdbcMetadataFactory.class).in(Scopes.SINGLETON);
```

It is an optional binder that you can override and use your custom factory.

Reply:
```
        newOptionalBinder(binder, JdbcMetadataFactory.class).setBinding().to(IgniteMetadataFactory.class).in(Scopes.SINGLETON);
```

Reply:
I got you, will try to use ``JdbcModule``